### PR TITLE
primers: iterators, parIters

### PR DESCRIPTION
**Pull request by psahabu**
@ben-albrecht:
* substantial overhaul to formatting
* wordsmithing for clarity

### CODE REVIEW

File: test/release/examples/primers/parIters.chpl [hunk 1]
```diff
@@ -238,26 +216,28 @@ iter count(param tag: iterKind, n: int, low: int=1)
     yield (zeroBasedIters,);
   }
 }
+
 //
 // As mentioned at the outset, this leader is fairly static and
 // simple.  More generally, a leader can introduce tasks more
 // dynamically, partition work between the tasks more dynamically,
-// etc.  See $CHPL_HOME/modules/standard/DynamicIters.chpl for
+// etc.  ``See $CHPL_HOME/modules/standard/DynamicIters.chpl`` for
```
Comment (ben-albrecht) [inline test/release/examples/primers/parIters.chpl]: See :mod:`DynamicIters:`

File: test/release/examples/primers/iterators.chpl [hunk 2]
```diff
@@ -2,48 +2,40 @@
...
    It also contains examples of the two kinds of parallel iteration:
-    data-parallel (forall), and
-    task-parallel (coforall).
+     * data-parallel (``forall``), and
+     * task-parallel (``coforall``).
 */
 
 //
-// fibonacci - generates the first n Fibonacci numbers
+// fibonacci
```
Comment (ben-albrecht) [inline test/release/examples/primers/iterators.chpl]: Applies to all sections:

We've been adding reference labels to section titles to support potential cross-referencing in the future.

e.g.

```
/*
.. primers-iterators-fibonacci:

fibonacci
---------
*/
```

File: test/release/examples/primers/parIters.chpl [hunk 3]
```diff
@@ -372,38 +351,35 @@ writeln("After reassigning A using parallel zippering and A as the leader:");
...
+// of changes planned for it such as interface improvements and better
+// error checking.  We'll update this primer as we improve these features.
 //
-//   As mentioned at the outset, our leader-follower scheme has a number
-//   of changes planned for it such as interface improvements and better
-//   error checking.  We'll update this primer as we improve these features.
-//
-
-
+// Appendix:
```
Comment (ben-albrecht) [inline test/release/examples/primers/parIters.chpl]: Maybe this should be a sub-section? Not sure.

File: test/release/examples/primers/iterators.chpl (hunk 2 above)
Comment (ben-albrecht) [inline test/release/examples/primers/iterators.chpl]: Also, might look nice to capitalize section titles

File: test/release/examples/primers/iterators.chpl [hunk 4]
```diff
@@ -183,34 +172,25 @@ writeln("Tree Data");
...
-// The coforall loop uses the serial version iterator
-// to spawn a separate task for each of the values it yields.
-// If you use coforall, you are asserting that the manipulations
-// done with each yielded value can be done in parallel
-// (i.e. in no particular order).
-//
-// All of the spawned tasks must complete before execution continues
+// the iterator is required to create parallel tasks and assign work to
+// them.  Such iterators are a fairly big topic and are described in
+// detail in :ref:`primers-parIters`.
```
Comment (ben-albrecht) [inline test/release/examples/primers/iterators.chpl]: ``:ref:`Parallel Iterators Primer <primers-parIters>` ``

File: test/release/examples/primers/parIters.chpl [hunk 5]
```diff
@@ -1,67 +1,68 @@
...
   forall loops in Chapel over data structures or iterators.
   Standalone parallel iterators are used to implement non-zippered
   forall loops when they exist, falling back to leader-follower when
-  standalone is not available. We expect the parallel iterator
-  interface to change and improve over time, so this should be
-  considered a snapshot of their use in the current implementation.
+  standalone is not available. 
 
   For a more thorough introduction to leader-follower iterators,
   refer to our PGAS 2011 paper, "User-Defined Parallel Zippered
```
Comment (ben-albrecht) [inline test/release/examples/primers/parIters.chpl]: Maybe make this a URL to http://pgas11.rice.edu/papers/ChamberlainEtAl-Chapel-Iterators-PGAS11.pdf ?

Use the style that allows you to put the URL at the bottom of the file, see [here](http://docutils.sourceforge.net/docs/user/rst/quickref.html#hyperlink-targets)

File: test/release/examples/primers/parIters.chpl (hunk 3 above)
Comment (psahabu) [inline test/release/examples/primers/parIters.chpl]: It looked rather strange to me, as it appears in the sidebar when it really doesn't matter at all. I'll pass if that's alright.

**Pull request merged**